File: .ProgramingBasics/DrawingFiguresWithLoops/10.Diamond/src/Diamond.java
import java.text.MessageFormat;
import java.util.Scanner;

public class Diamond {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        int dashes = (n - 1) / 2;
        int middleDashes = 1;
        int stars = 1;
        int oddOrEven = 1;

        if (n % 2 == 0) {
            stars = 2;
            middleDashes = 2;
            oddOrEven = 0;
        }

        if (n == 1) {
            System.out.println("*");
        }
        else if (n == 2) {
            System.out.println("**");
        }
        else {
            for (int i = 0; i < n / 2 + oddOrEven; i++) {
                if (i == 0) {
                    System.out.println(
                        MessageFormat.format("{0}{1}{0}",
                        repeatStr("-", (n - 1) / 2), repeatStr("*", stars))
                    );
                }
                else {
                    dashes--;
                    System.out.println(
                        MessageFormat.format("{0}*{1}*{0}",
                        repeatStr("-", dashes), repeatStr("-", middleDashes))
                    );
                    middleDashes += 2;
                }
            }
            dashes = 0;
            middleDashes -= 2;

            for (int i = 0; i < (n + 1) / 2 - 2; i++) {
                dashes++;
                middleDashes -= 2;
                System.out.println(
                    MessageFormat.format("{0}*{1}*{0}",
                    repeatStr("-", dashes), repeatStr("-", middleDashes))
                );
            }

            System.out.println(
                MessageFormat.format("{0}{1}{0}",
                    repeatStr("-", (n - 1) / 2), repeatStr("*", stars))
            );
        }
    }

    private static String repeatStr(String str, int count) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < count; i++) {
            sb.append(str);
        }

        return sb.toString();
    }
}

File: ProgramingBasics/DrawingFiguresWithLoops/10.Diamond/10.Diamond/Diamond.cs
using System;

namespace _10.Diamond
{
    public class Diamond
    {
        public static void Main(string[] args)
        {
            var n = int.Parse(Console.ReadLine());
            var dashes = (n - 1) / 2;
            var middleDashes = 1;
            var stars = 1;
            var oddOrEven = 1;

            if (n % 2 == 0)
            {
                stars = 2;
                middleDashes = 2;
                oddOrEven = 0;
            }

            if (n == 1)
            {
                Console.WriteLine("*");
            }
            else if (n == 2)
            {
                Console.WriteLine("**");
            }
            else
            {
                for (var i = 0; i < n / 2 + oddOrEven; i++)
                {
                    if (i == 0)
                    {
                        Console.WriteLine("{0}{1}{0}", new string('-', (n - 1) / 2), new string('*', stars));
                    }
                    else
                    {
                        dashes--;
                        Console.WriteLine("{0}*{1}*{0}", new string('-', dashes), new string('-', middleDashes));
                        middleDashes += 2;
                    }
                }

                dashes = 0;
                middleDashes -= 2;

                for (var i = 0; i < (n + 1) / 2 - 2; i++)
                {
                    dashes++;
                    middleDashes -= 2;
                    Console.WriteLine("{0}*{1}*{0}", new string('-', dashes), new string('-', middleDashes));
                }

                Console.WriteLine("{0}{1}{0}", new string('-', (n - 1) / 2), new string('*', stars));
            }
        }
    }
}